.github/workflows/ubuntu.yml

```diff
@@ -40,7 +40,7 @@ jobs:
       run: ${{ matrix.compiler }} --version
     - name: 'Configure (meson)'
       if: ${{ (matrix.build == 'meson') }}
-      run: meson setup ./builddir --prefix=/usr
+      run: meson setup ./builddir --prefix=/usr -D werror=true
```

Comment:
AFAICT this does now work with MSVC since the `-Werror` style is gcc specific. In other words even if used with clang (in msvc mode) this won't trigger.

So I would rather stay consistent across the meson board and use the same flag. Considering the shift away from autotools and the (pre-existing bugs in autotools) I seriously won't bother, and would rather use the proper style.

Reply:
On visual studio style compilers (MSVC, clang-cl) the Meson setting `-Dwerror=true` is translated to the compiler flag `/WX`. This is a good proof that Meson should indeed be in charge of this, but if you need further proof, there are a variety of other things it could be, depending both on the compiler, and the programming language:
- `--fatal-warnings`
- `--emit_warnings_as_errors`
- `-D warnings`
- `-w`
- `-Werror=cross-execution-space-call,deprecated-declarations,reorder`
- `-warnaserror`
- `-change_message=error`


I suppose however, that you're not using ccrx, ti, or xc16 branded C compilers in CI, so the only practical issue you run into for portability, is indeed MSVC and clang-cl.

Reply:
If meson has exposed a generalized option across a board to handle a setting which is different across different compilers - that is a good thing. However, I do not see why I can not use a practice to specify additional compiler options (thru `CFLAGS`) which I used for 2 decades and which so far has worked for me? Especially why I can't use it for the particular CI case with particular compilers (gcc/clang) running on particular OS (ubuntu)? Moreover, if that's the way how most people would specify additional compiler flags, it does make sense to me to check in CI that project correctly handles such external specification (and I did see far too often when projects do break this convention). That's the reason I personally prefer to leave CFLAGS for our ubuntu CI case with gcc/clang.

By the way, a question. What's meson "recommended" way to specify arbitrary compiler/linker flags? Is that thru CFLAGS or there is some other better way?

Reply:
> However, I do not see why I can not use a practice to specify additional compiler options (thru `CFLAGS`) which I used for 2 decades and which so far has worked for me?

You can, and Meson will respect that by assuming you know exactly which flags you need.

But... the flags you specified do not work on MSVC. :)



> By the way, a question. What's meson "recommended" way to specify arbitrary compiler/linker flags? Is that thru CFLAGS or there is some other better way?

Meson respects the standard CFLAGS for arbitrary compiler/linker flags.

Some things, Meson has builtin support for via Meson options, which you can use as well -- and if Meson has a builtin option for it, that option will be cross-platform, rather than having to specify one set of CFLAGS for GCC and another set for MSVC.

However, you *absolutely* can specify `CFLAGS=-Werror` on GCC and `CFLAGS=/WX` on MSVC, if you prefer.

Reply:
Must admit I did not expect such a trivial change to cause that much of a discussion. Above all - thanks Eli for providing insights from meson's POV.

@dvrogozh onto your points:
 - meson will honour CFLAGS, so you can still use them but as mentioned that will not work for other people's use-cases
 - to meson or autotools - as hinted in the past, people and more importantly distributions want to move away from the latter. the fact that FreeBSD and Windows use the former is a clear indication, especially since autotools just works with them

The second is a yak shaving which should be left separate IMHO. Will address your other comment in a bit.